jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/StandardWarExplodedProcessor.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.war;
+
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.cli.ArtifactLayers;
+import com.google.cloud.tools.jib.cli.ArtifactProcessor;
+import com.google.cloud.tools.jib.plugins.common.ZipUtil;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.MoreFiles;
+import com.google.common.io.RecursiveDeleteOption;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+class StandardWarExplodedProcessor implements ArtifactProcessor {
+
+  private final Path warPath;
+  private final Path targetExplodedWarRoot;
+  private final Integer warJavaVersion;
+  private final AbsoluteUnixPath appRoot;
+
+  /**
+   * Constructor for {@link StandardWarExplodedProcessor}.
+   *
+   * @param warPath path to WAR file
+   * @param targetExplodedWarRoot path to exploded-war root
+   * @param warJavaVersion war java version
+   * @param appRoot the absolute path of the app on the container
+   */
+  StandardWarExplodedProcessor(
+      Path warPath, Path targetExplodedWarRoot, Integer warJavaVersion, AbsoluteUnixPath appRoot) {
+    this.warPath = warPath;
+    this.targetExplodedWarRoot = targetExplodedWarRoot;
+    this.warJavaVersion = warJavaVersion;
+    this.appRoot = appRoot;
+  }
+
+  @Override
+  public List<FileEntriesLayer> createLayers() throws IOException {
+    // Clear the exploded-jar root first
+    if (Files.exists(targetExplodedWarRoot)) {
+      MoreFiles.deleteRecursively(targetExplodedWarRoot, RecursiveDeleteOption.ALLOW_INSECURE);
+    }
+
+    ZipUtil.unzip(warPath, targetExplodedWarRoot, true);
+    Predicate<Path> isFile = Files::isRegularFile;
+    Predicate<Path> isInWebInfLib =
+        path -> path.startsWith(targetExplodedWarRoot.resolve("WEB-INF").resolve("lib"));
+    Predicate<Path> isSnapshot = path -> path.getFileName().toString().contains("SNAPSHOT");
+
+    // Non-snapshot layer
+    Predicate<Path> isInWebInfLibAndIsNotSnapshot = isInWebInfLib.and(isSnapshot.negate());
+    Predicate<Path> nonSnapshotPredicate = isInWebInfLibAndIsNotSnapshot;
```

Comment:
At this point, I think we should just remove the variable `nonSnapshotPredicate`.